### ISSUE TITLE
False positive for `undefined-loop-variable` with `enumerate()`

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#6593]
### Bug description
```python
for i, num in enumerate(range(3)):
    pass
print(i, num)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:3: [W0631(undefined-loop-variable), ] Using possibly undefined loop variable 'i'
a.py:3: [W0631(undefined-loop-variable), ] Using possibly undefined loop variable 'num'

------------------------------------------------------------------
Your code has been rated at 3.33/10 (previous run: 0.00/10, +3.33)
```

### Expected behavior

no msg. Notice that without enumerate, pylint doesn't emit messages.

### Pylint version

```shell
pylint 2.14.0-b1
astroid 2.12.0-dev0
Python 3.10.2 (v3.10.2:a58ebcc701, Jan 13 2022, 14:50:16) [Clang 13.0.0 (clang-1300.0.29.30)]
```
